service/cert-signer/cert_signer.go

```diff
@@ -101,6 +133,41 @@ func (cs *certSigner) Issue(config spec.IssueConfig) (spec.IssueResponse, error)
 	return newIssueResponse, nil
 }
 
-func (cs *certSigner) SignedPath(clusterID string) string {
-	return fmt.Sprintf("pki-%s/issue/role-%s", clusterID, clusterID)
+func (cs *certSigner) SignedPath(clusterID string, organizations string) string {
+	return fmt.Sprintf("pki-%s/issue/%s", clusterID, roleName(clusterID, organizations))
+}
+
+func roleName(clusterID string, organizations string) string {
```

Comment:
It is a bit confusing because of the name. But this computes the role name based on the cluster and list of organizations we want to issue a cert for. This belongs to the cert signer.

The cert signer knows what the name of the role should be, because it owns the knowledge of how to issue the cert for the cn and the org.

The role service is only concerned with how to create roles, given a name. It does not, and should not know about this domain specific `clusterID` and `organizations` concepts.